platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/ParameterProvider.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright (C) 2023 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.swirlds.platform;
+
+import edu.umd.cs.findbugs.annotations.Nullable;
+
+/**
+ * A utility class for obtaining the command line parameters when launching the platform using Browser.java.
+ *
+ * @deprecated this class will be removed at a future date, use the configuration engine for app config when launching
+ * from the browser
+ */
+@Deprecated(forRemoval = true)
+public final class ParameterProvider {
+
+    private static String[] parameters;
+
+    private ParameterProvider() {}
+
+    /**
+     * Set the command line parameters.
+     *
+     * @param parameters the command line parameters
+     */
+    static synchronized void setParameters(@Nullable final String[] parameters) {
```

Comment:
Since we mutate values here I would like a singleton much more than having all the methods static.

Reply:
converted to a singleton